### PR TITLE
NonactivatedUser.inc: Use modern DPDatabase::query, escaping, sprintf

### PR DESCRIPTION
**Pull request by bpfoley**
Unit tests pass (yay for unit tests)

### CODE REVIEW

File: pinc/NonactivatedUser.inc [hunk 1]
```diff
@@ -80,8 +80,7 @@ class NonactivatedUser
             FROM non_activated_users
             WHERE $field=$escaped_value
         ";
-
-        $result = mysqli_query(DPDatabase::get_connection(), $sql);
+        $result = DPDatabase::query($sql, FALSE);
```
Comment (chrismiceli) [inline pinc/NonactivatedUser.inc:83]: We should probably get rid of the error handling here and let the ::query function take care of it.

Comment (bpfoley) [reply pinc/NonactivatedUser.inc]: Done. Nothing was relying on the old exception type anyway.

**Comment by cpeel** [comment]
I'm going to ask you to start testing these yourself before you put them up for code review:
```
Notice: Undefined variable: ql in /home/cpeel/u-cpeel-dp/pinc/NonactivatedUser.inc on line 199

Warning: mysqli_query(): Empty query in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc on line 69

Fatal error: Uncaught DBQueryError: An error occurred during a database query and has been logged. in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc:76 Stack trace: #0 /home/cpeel/u-cpeel-dp/pinc/NonactivatedUser.inc(199): DPDatabase::query(NULL) #1 /home/cpeel/u-cpeel-dp/accounts/activate.php(122): NonactivatedUser->delete() #2 {main} thrown in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc on line 76
```

**Comment by bpfoley** [comment]
> I'm going to ask you to start testing these yourself before you put them up for code review:
> 
> ```
> Notice: Undefined variable: ql in /home/cpeel/u-cpeel-dp/pinc/NonactivatedUser.inc on line 199
> 
> Warning: mysqli_query(): Empty query in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc on line 69
> 
> Fatal error: Uncaught DBQueryError: An error occurred during a database query and has been logged. in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc:76 Stack trace: #0 /home/cpeel/u-cpeel-dp/pinc/NonactivatedUser.inc(199): DPDatabase::query(NULL) #1 /home/cpeel/u-cpeel-dp/accounts/activate.php(122): NonactivatedUser->delete() #2 {main} thrown in /home/cpeel/u-cpeel-dp/pinc/DPDatabase.inc on line 76
> ```

D'oh! I was lulled into a false sense of security by the unit tests. I've added a unit test for delete now.

**Pull request merged**